File: Python/registratorecassa/registratorecassa.py
costofinale = 0
pagamento = False

while True:
	print('''
	Vuoi aggiungere un articolo alla cassa?\n
	1. Si (y)
	2. No (n)
	''')
	risposta = input('Riposta = ')

	if(risposta == 'y'):
		costoarticolo = float(input('\nPrezzo articolo = EUR '))
		numeroarticoli = float(input('Quantità articolo: '))
		scontoarticolo = float(input('Sconto del prodotto = % '))

		costoarticolo = costoarticolo * numeroarticoli
		costoscontato = (costoarticolo * scontoarticolo) / 100
		costoscontato = costoarticolo - costoscontato
		costofinale = costofinale + costoscontato
	elif(risposta == 'n'):
		break

print('\nLista articoli terminata!')
print('Totale spesa = ' + str(costofinale) + '\n')

if(costofinale > 0):
	pagamento = float(input('Effettuare il pagamento: EUR '))
if(pagamento > costofinale):
	resto = pagamento - costofinale
	print('Il suo resto è di EUR ' + str(resto))

File: C/school/2018/RegistratoreCassa/RegistratoreCassa.c
/*Scrivere un algoritmo tale che simuli il funzionamento
di un registratore di cassa per il calcolo di uno scontrino.
Per ogni articolo venduto, bisognerà inserire quantità, prezzo
unitario e % di sconto, e poi battere un tasto per dire che si
intende inserire un nuovo articolo, oppure un altro tasto per
dire che è finita la lista degli articoli. Terminata la lista
degli articoli il simulatore dovrà mostrare il totale calcolato,
chiedere l'importo versato e calcolare e mostrare il resto da dare
al cliente.
*/

#include <stdio.h>
#include <stdlib.h>

int main()
{
  char r;
  int itemc,itemq,itemt,sale,p=NULL,prest;
  int c=1,s=0;

  system("cls");
  printf("*Registratore Cassa*\n");

  do
  {
    printf("\nVuoi aggiungere un %d\xF8 articolo alla cassa?\n",c);
    printf("\n1. Si(y)\n2. No(n)\n");
    printf("\nRisposta = ");scanf(" %c",&r);

    if(r=='y')
    {
      c++;
      printf("\nPrezzo articolo = EUR ");scanf("%d",&itemc);
      printf("Quantit\x85 articolo = ");scanf("%d",&itemq);
      printf("Sconto del prodotto = %%");scanf("%d",&sale);
      itemc=itemc*itemq;
      itemt=(itemc*sale)/100;itemt=itemc-itemt;
      s=s+itemt;
    }
  }while(r!='n');
  
  printf("\nLista articoli terminata!\n");
  printf("Totale Spesa = %d\n",s);
  
  if(s>0)
  {
    printf("\nEffettuare il pagamento: EUR ");scanf("%d",&p);
  }

  if(p>s)
  {
    prest=p-s;
    printf("Il suo resto \x8A di EUR %d\n",prest);
  }

  printf("\n");
  system("pause");
}